plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/procedure/TestIcebergAddFilesProcedure.java

```diff
@@ -0,0 +1,576 @@
+        assertUpdate("DROP TABLE iceberg.tpch." + icebergTableName);
+    }
+
+    @Test
+    void testAddFilesDifferentFileFormat()
+    {
+        testAddFilesDifferentFileFormat("PARQUET", "ORC");
+        testAddFilesDifferentFileFormat("PARQUET", "AVRO");
+        testAddFilesDifferentFileFormat("ORC", "PARQUET");
+        testAddFilesDifferentFileFormat("ORC", "AVRO");
+        testAddFilesDifferentFileFormat("AVRO", "PARQUET");
+        testAddFilesDifferentFileFormat("AVRO", "ORC");
+    }
+
+    private void testAddFilesDifferentFileFormat(String hiveFormat, String icebergFormat)
+    {
+        String hiveTableName = "test_add_files_" + randomNameSuffix();
+        String icebergTableName = "test_add_files_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE iceberg.tpch." + icebergTableName + " WITH (format = '" + icebergFormat + "') AS SELECT 1 x", 1);
+        assertUpdate("CREATE TABLE hive.tpch." + hiveTableName + " WITH (format = '" + hiveFormat + "') AS SELECT 2 x", 1);
+
+        assertUpdate("ALTER TABLE " + icebergTableName + " EXECUTE add_files_from_table('tpch', '" + hiveTableName + "')");
+
+        assertQuery("SELECT * FROM " + icebergTableName, "VALUES 1, 2");
+
+        assertUpdate("DROP TABLE hive.tpch." + hiveTableName);
+        assertUpdate("DROP TABLE iceberg.tpch." + icebergTableName);
+    }
+
+    @Test
+    void testAddFilesAcrossSchema()
+    {
+        String hiveSchemaName = "test_schema" + randomNameSuffix();
+        String hiveTableName = "test_add_files_" + randomNameSuffix();
+        String icebergTableName = "test_add_files_" + randomNameSuffix();
+
+        assertUpdate("CREATE SCHEMA hive." + hiveSchemaName);
+
+        assertUpdate("CREATE TABLE iceberg.tpch." + icebergTableName + " WITH (format = 'PARQUET') AS SELECT 1 x", 1);
+        assertUpdate("CREATE TABLE hive." + hiveSchemaName + "." + hiveTableName + " WITH (format = 'ORC') AS SELECT 2 x", 1);
+
+        assertUpdate("ALTER TABLE tpch." + icebergTableName + " EXECUTE add_files_from_table('" + hiveSchemaName + "', '" + hiveTableName + "')");
+
+        assertQuery("SELECT * FROM iceberg.tpch." + icebergTableName, "VALUES 1, 2");
+
+        assertUpdate("DROP TABLE iceberg.tpch." + icebergTableName);
+        assertUpdate("DROP SCHEMA hive." + hiveSchemaName + " CASCADE ");
+    }
+
+    @Test
+    void testAddFilesTypeMismatch()
+    {
+        String hiveTableName = "test_add_files_" + randomNameSuffix();
+        String icebergTableName = "test_add_files_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE iceberg.tpch." + icebergTableName + " WITH (format = 'ORC') AS SELECT '1' x", 1);
+        assertUpdate("CREATE TABLE hive.tpch." + hiveTableName + " WITH (format = 'ORC') AS SELECT 2 x", 1);
+
+        assertQueryFails(
+                "ALTER TABLE " + icebergTableName + " EXECUTE add_files_from_table('tpch', '" + hiveTableName + "')",
+                "Expected target 'string' type, but got source 'int' type");
```

Comment:
optional: This doesn't feel very user friendly for debugging purposes.

Reply:
Exactly, I included the column name like `Target '%s' column is '%s' type, but got source '%s' type`.